src/commands/explorer/groupBy.ts

```diff
@@ -3,29 +3,39 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
-import { IActionContext } from "@microsoft/vscode-azext-utils";
+import { uiUtils } from "@microsoft/vscode-azext-azureutils";
+import { IActionContext, nonNullProp, subscriptionExperience } from "@microsoft/vscode-azext-utils";
+import { AzureSubscription } from "@microsoft/vscode-azext-utils/hostapi.v2";
 import { QuickPickItem } from "vscode";
+import { ext } from "../../extensionVariables";
+import { createResourceClient } from "../../utils/azureClients";
 import { localize } from "../../utils/localize";
 import { settingUtils } from "../../utils/settingUtils";
+import { createSubscriptionContext } from "../../utils/v2/credentialsUtils";
 
 export function buildGroupByCommand(setting: string) {
     return (context: IActionContext): Promise<void> => groupBy(context, setting);
 }
 
 async function groupBy(context: IActionContext, setting: string): Promise<void> {
     if (setting === 'armTag') {
-        const tag = await context.ui.showQuickPick(getQuickPicks(context), {
-            placeHolder: localize('groupByArmTagKey', 'Select the tag key to group by...')
+        const subscription = await subscriptionExperience(context, ext.v2.api.resources.azureResourceTreeDataProvider);
```

Comment:
Should we move this to `getQuickPicks`? Put all the async code in one spot and all.

Reply:
Since the subscription experience involves showing a quick pick, I think it needs to (and should) stay out of `getQuickPicks` because while `getQuickPicks` is being awaited, we show loading state in the UI.